showcase/src/app/authentication.config.ts

```diff
@@ -0,0 +1,111 @@
+import { StarkUser } from "@nationalbankbelgium/stark-core";
+
+export function getAuthenticationHeaders(user: StarkUser): Map<string, string> {
+	const devAuthentication: StarkDevAuthentication = {
+		roleSeparator: "^",
+		descriptionSeparator: "/",
+		defaults: {
+			language: "F",
+			workpost: "XXX",
+			referenceNumber: "00000"
+		}
+	};
+
+	// set a default language if not known
+	const languageHeaderValue: string = user && user.language ? user.language : devAuthentication.defaults.language;
+	// set a default workpost if not known
+	const workpost: string = user && user.workpost ? user.workpost : devAuthentication.defaults.workpost;
+	// set a default reference number if not known
+	const referenceNumber: string = user && user.referenceNumber ? user.referenceNumber : devAuthentication.defaults.referenceNumber;
+
+	const descriptionHeaderValue: string = referenceNumber + devAuthentication.descriptionSeparator + workpost;
+
+	let rolesHeaderValue: string = "";
+	if (user && user.roles) {
+		rolesHeaderValue = user.roles.join(devAuthentication.roleSeparator);
+	}
+
+	const devAuthenticationHeaders: Map<string, string> = new Map<string, string>();
+
+	if (user) {
+		devAuthenticationHeaders.set(StarkHttpHeaders.NBB_USER_NAME, user.username);
+		devAuthenticationHeaders.set(StarkHttpHeaders.NBB_FIRST_NAME, user.firstName);
+		devAuthenticationHeaders.set(StarkHttpHeaders.NBB_LAST_NAME, user.lastName);
+		if (user.email) {
+			devAuthenticationHeaders.set(StarkHttpHeaders.NBB_MAIL, user.email);
+		}
+	}
+	devAuthenticationHeaders.set(StarkHttpHeaders.NBB_LANGUAGE, languageHeaderValue);
+	devAuthenticationHeaders.set(StarkHttpHeaders.NBB_DESCRIPTION, descriptionHeaderValue);
+	devAuthenticationHeaders.set(StarkHttpHeaders.NBB_ROLES, rolesHeaderValue);
+
+	return devAuthenticationHeaders;
+}
+
+/**
+ * NBB's Java back-ends are usually configured to expect pre-authentication; that is the fact that when a request comes in, if should have been authenticated before by "something else".
+ * The way it works is that each request must include NBB-specific HTTP headers along with each request.
+ * When you are working against a development environment (e.g., back-end running on your own machine), the infrastructure is not there, so we must simulate it.
+ */
+interface StarkDevAuthentication {
```

Comment:
Maybe this should not be have the "Stark" prefix anymore since it is app specific, what about prefixing them with "App"?